crates/ruff_linter/src/rules/flake8_trio/rules/timeout_without_await.rs

```diff
@@ -0,0 +1,108 @@
+use ruff_python_ast::visitor::{walk_expr, Visitor};
+use ruff_python_ast::{Expr, ExprAwait, ExprCall, StmtWith, WithItem};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for trio functions that should contain await but don't.
+///
+/// ## Why is this bad?
+///
+/// Some trio context managers, such as 'trio.fail_after' and
+/// 'trio.move_on_after', have no impact when there is no await statement in them.
+///
+/// ## Example
+/// ```python
+/// async def f():
+///     with trio.move_on_after(2):
+///         do_something()
+/// ```
+///
+/// Use instead:
+/// ```python
+/// async def f():
+///     with trio.move_on_after(2):
+///         do_something()
+///         await awaitable()
+/// ```
+#[violation]
+pub struct TimeoutWithoutAwait {
```

Comment:
I think we should prefix all these rules with `Trio`, like `TrioTimeoutWithoutAwait`, similar to our Pytest rules.